### PR TITLE
Example forecaster for modeltools

### PR DESCRIPTION
**Pull request by sgsmob**
Adds an example forecaster to `modeltools` that demonstrates how to use the `modeltools` functions to construct a forecaster with `quantgen`.

### CODE REVIEW

**Comment by sgsmob** [comment]
@dajmcdon This is my example forecaster

File: R-packages/modeltools/R/example_forecaster.R [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+#'                                      start_date = as.Date("2021-01-01"),
+#'                                    ),
+#'                                    as.Date("2021-03-01"),
+#'                                    "epiweek")
+#'
+#' @export
+example_forecaster <- function(df_list, forecast_date) {
+    ahead <- 1:4
+    lags <- c(0, -1, -2, -3, -7, -14)
+    quantiles <- modeltools::covidhub_probs()
```
Comment (dajmcdon) [inline R-packages/modeltools/R/example_forecaster.R:21]: Is `covidhub_probs` a function or data? See line 115 in `quantgen.R`.

File: R-packages/modeltools/R/example_forecaster.R [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    covariates <- covidcast::aggregate_signals(df_list, dt = lags, format = "wide")
+    covariates_with_response <- add_response_columns(covariates,
+                                                     "jhu-csse_confirmed_incidence_num",
+                                                     forecast_date,
+                                                     "epiweek",
+                                                     ahead)
+
+    results <- list()
+    for (a in ahead) {
+        mats <- create_train_and_predict_matrices(covariates_with_response, a, 14)
```
Comment (dajmcdon) [inline R-packages/modeltools/R/example_forecaster.R:32]: the `14` here is the training window size? I'm wondering if this, `ahead` and `lags` shouldn't be arguments to the function?

File: R-packages/modeltools/R/quantgen.R [hunk 3]
```diff
@@ -109,7 +109,7 @@
 #' @importFrom dplyr filter select pull summarize between bind_cols
 #' @importFrom tidyr pivot_longer
 #' @export
-quantgen_forecaster = function(df, forecast_date, signals, incidence_period,
+quantgen_forecaster = function(df_list, forecast_date, signals, incidence_period,
```
Comment (dajmcdon) [inline R-packages/modeltools/R/quantgen.R:112]: This signature won't work with the new `evalcast`. Unless you pass some of these through the `forecaster_args()` list.

Comment (dajmcdon) [reply R-packages/modeltools/R/quantgen.R:112]: Most of it is actually unnecessary though.

**Review by dajmcdon — CHANGES_REQUESTED** [review]

File: R-packages/modeltools/R/quantgen.R (hunk 3 above)
Comment (sgsmob) [reply R-packages/modeltools/R/quantgen.R:112]: The plan is to make it all compliant with the new evalcast although I realize that they are not visible from one another yet.

File: R-packages/modeltools/R/example_forecaster.R (hunk 1 above)
Comment (sgsmob) [reply R-packages/modeltools/R/example_forecaster.R]: Fixed

File: R-packages/modeltools/R/example_forecaster.R (hunk 2 above)
Comment (sgsmob) [reply R-packages/modeltools/R/example_forecaster.R]: Done.

**Review by dajmcdon — APPROVED** [review]

**Pull request merged**